python/dgl/distributed/rpc.py

```diff
@@ -10,12 +10,159 @@
 
 REQUEST_CLASS_TO_SERVICE_ID = {}
 RESPONSE_CLASS_TO_SERVICE_ID = {}
 SERVICE_ID_TO_PROPERTY = {}
 
+def read_ip_config(filename):
+    """Read network configuration information of server from file.
+
+    The format of configuration file should be:
+
+        [ip] [base_port] [server_count]
+
+        172.31.40.143 30050 2
+        172.31.36.140 30050 2
+        172.31.47.147 30050 2
+        172.31.30.180 30050 2
+
+    Note that, DGL supports multiple backup servers that shares data with each others
+    on the same machine via shared-memory tensor. The server_count should be >= 1. For example,
+    if we set server_count to 5, it means that we have 1 main server and 4 backup servers on
+    current machine. Note that, the count of server on each machine can be different.
+
+    Parameters
+    ----------
+    filename : str
+        Path of IP configuration file.
+
+    Returns
+    -------
+    dict
+        server namebook. The format is: {[server_id]:[machine_id, ip, port, group_count]}
+
+        e.g.,
+
+          {0:[0, '172.31.40.143', 30050, 2],
+           1:[0, '172.31.40.143', 30051, 2],
+           2:[1, '172.31.36.140', 30050, 2],
+           3:[1, '172.31.36.140', 30051, 2],
+           4:[2, '172.31.47.147', 30050, 2],
+           5:[2, '172.31.47.147', 30051, 2],
+           6:[3, '172.31.30.180', 30050, 2],
+           7:[3, '172.31.30.180', 30051, 2]}
+    """
+    assert len(filename) > 0, 'filename cannot be empty.'
+    server_namebook = {}
+    try:
+        server_id = 0
+        machine_id = 0
+        lines = [line.rstrip('\n') for line in open(filename)]
+        for line in lines:
+            ip_addr, port, server_count = line.split(' ')
+            for s_count in range(int(server_count)):
+                server_namebook[server_id] = \
+                [int(machine_id), ip_addr, int(port)+s_count, int(server_count)]
+                server_id += 1
+            machine_id += 1
+    except:
+        print("Error: data format on each line should be: [ip] [base_port] [server_count]")
+    return server_namebook
+
+def create_sender(msg_queue_size, net_type):
+    """Create rpc sender of this process.
+
+    Parameters
+    ----------
+    msg_queue_size : int
+        Size (bytes) of network queue buffer.
+    net_type : str
+        networking type, e.g., 'socket' or 'mpi' (do not support yet).
+    """
+    _CAPI_DGLRPCCreateSender(int(msg_queue_size), net_type)
+
+def create_receiver(msg_queue_size, net_type):
+    """Create rpc receiver of this process.
+
+    Parameters
+    ----------
+    msg_queue_size : int
+        Size (bytes) of network queue buffer.
+    net_type : str
+        networking type, e.g., 'socket' or 'mpi' (do not support yet).
+    """
+    _CAPI_DGLRPCCreateReceiver(int(msg_queue_size), net_type)
+
+def get_sender():
+    """Get rpc sender of this process.
+
+    Returns
+    -------
+    c handler
+        sender handler
+    """
+    return _CAPI_DGLRPCGetSender()
+
+def get_receiver():
```

Comment:
Do we need `get_sender` and `get_receiver`? They can be retrieved directly in C++ right?

Reply:
Yes, I think we can remove these two API.